sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -86,13 +87,50 @@ case class DescribeDatabase(
     extended: Boolean)(sql: String)
   extends NativeDDLCommand(sql) with Logging
 
+/**
+ * The DDL command that creates a function.
+ * alias: the class name that implements the created function.
+ * resources: Jars, files, or archives which need to be added to the environment when the function
+ *            is referenced for the first time by a session.
+ * isTemp: indicates if it is a temporary function.
+ */
 case class CreateFunction(
     databaseName: Option[String],
     functionName: String,
     alias: String,
     resources: Seq[(String, String)],
```

Comment:
resources is not used here? I think we need to change `CatalogFunction` to include resources so we can propagate that to Hive properly.